### PR TITLE
usm: classification: Fixed AMQP publish flakiness

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Improves classification by AMQP.

### Motivation

Publish has no response, so added classification by close-reply and close-ok messages.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/amqp-helpers.h [hunk 1]
```diff
@@ -30,28 +29,47 @@ static __always_inline bool is_amqp(const char* buf, __u32 buf_size) {
        return false;
     }
 
-    uint8_t frame_type = buf[0];
+    __u8 frame_type = buf[0];
```
Comment (DanielLavie) [inline pkg/network/ebpf/c/protocols/amqp-helpers.h:32]: Why are we choosing to prefer the `__ux` type over the `uintx_t` type?

Comment (guyarb) [reply pkg/network/ebpf/c/protocols/amqp-helpers.h:32]: sticking with the wide convention in the project
having mixed types only causes an un-readable code

File: pkg/network/tracer/tracer_classification_test.go [hunk 2]
```diff
@@ -54,6 +54,7 @@ type testContext struct {
 }
 
 func setupTracer(t *testing.T, cfg *config.Config) *Tracer {
+	cfg.BPFDebug = true
```
Comment (DanielLavie) [inline pkg/network/tracer/tracer_classification_test.go:57]: Why force it here? one can control it by setting the env variable: `BPF_DEBUG=true`

Comment (guyarb) [reply pkg/network/tracer/tracer_classification_test.go:57]: oops, removed

**Review by DanielLavie — APPROVED** [review]

**Pull request merged**